L1Trigger/TrackTrigger/src/HitPatternHelper.cc

```diff
@@ -0,0 +1,274 @@
+      layerIds_.push_back(std::make_pair(sm.layerId(), sm.barrel()));
+    }
+    sort(layerIds_.begin(), layerIds_.end(), smallerID);
+    layerIds_.erase(unique(layerIds_.begin(), layerIds_.end(), equalID), layerIds_.end());  //Keep only unique layerIds
+    // Converting tmtt::KFbase::layerMap to a format that is acceptatble by HitPatternHelper
+    for (int i = 0; i < nEtaRegions_; i++) {
+      for (int j = 0; j < (int)layerIds_.size(); j++) {
+        int layer = nKalmanLayers_;
+        if (layerIds_[j].second) {
+          layer = tmtt::KFbase::layerMap[i][tmtt::TrackerModule::calcLayerIdReduced(layerIds_[j].first)].first;
+        } else {
+          layer = tmtt::KFbase::layerMap[i][tmtt::TrackerModule::calcLayerIdReduced(layerIds_[j].first)].second;
+        }
+        if (layer < nKalmanLayers_) {
+          layermap_[i][layer].push_back(layerIds_[j].first);
+        }
+      }
+    }
+  }
+
+  HitPatternHelper::HitPatternHelper(const Setup* setup, int hitpattern, double cot, double z0)
+      : hitpattern_(hitpattern),
+        numExpLayer_(0),
+        numMissingLayer_(0),
+        numMissingPS_(0),
+        numMissing2S_(0),
+        numPS_(0),
+        num2S_(0),
+        numMissingInterior1_(0),
+        numMissingInterior2_(0),
+        cot_(cot),
+        z0_(z0),
+        setup_(setup),
+        binary_(11, 0),
+        hphDebug_(setup_->hphDebug()),
+        useNewKF_(setup_->useNewKF()),
+        chosenRofZ_(setup_->chosenRofZ()),
+        deltaTanL_(setup_->deltaTanL()),
+        etaRegions_(setup_->etaRegions()),
+        nKalmanLayers_(setup_->nKalmanLayers()),
+        layermap_(setup_->layermap()) {
+    //Calculating eta sector based on cot and z0
+    float kfzRef = z0_ + chosenRofZ_ * cot_;
+    int kf_eta_reg = 0;
+    for (int iEtaSec = 1; iEtaSec < ((int)etaRegions_.size() - 1); iEtaSec++) {  // Doesn't apply eta < 2.4 cut.
+      float etaMax = etaRegions_[iEtaSec];
+      float zRefMax = chosenRofZ_ / tan(2. * atan(exp(-etaMax)));
+      if (kfzRef > zRefMax) {
+        kf_eta_reg = iEtaSec;
+      }
+    }
+    etaSector_ = kf_eta_reg;
+    if (kf_eta_reg < ((int)etaRegions_.size() - 1) / 2) {
+      kf_eta_reg = ((int)etaRegions_.size() - 1) / 2 - 1 - kf_eta_reg;
+    } else {
+      kf_eta_reg = kf_eta_reg - (int)(etaRegions_.size() - 1) / 2;
+    }
+    //Looping over sensor modules to make predictions on which layers particles are expected to hit
+    for (const tt::SensorModule& sm : setup_->sensorModules()) {
+      double d = (z0_ - sm.z() + sm.r() * cot_) / (sm.cosTilt() - sm.sinTilt() * cot_);
+      double d_p =
+          (z0_ - sm.z() + sm.r() * (cot_ + deltaTanL_ / 2)) / (sm.cosTilt() - sm.sinTilt() * (cot_ + deltaTanL_ / 2));
+      double d_m =
+          (z0_ - sm.z() + sm.r() * (cot_ - deltaTanL_ / 2)) / (sm.cosTilt() - sm.sinTilt() * (cot_ - deltaTanL_ / 2));
+      //      if (!(abs(d_p) < sm.numColumns() * sm.pitchCol() / 2. && abs(d_m) < sm.numColumns() * sm.pitchCol() / 2.))
```

Comment:
please delete the out-commented lines of code (unless there is a particular reason they are kept?)

Reply:
Done